### DISCUSSION TITLE
Zsh ou bash?

### DISCUSSION THREAD

**Discussion by paulinhoerry** [frontendbr/forum#394] [Discussão]
Me peguei esses dias pensando, O Zsh trás um impacto maior à o Bash em um ambiente para desenvolvimento front-end? Qual vocês usam? Por quê?

**Comment by felipefialho** [comment]
Desde que comecei a usar o zsh não parei mais. Nunca senti perda de desempenho nos ambientes até agora.

**Comment by Kirmayrtomaz** [comment]
Nunca tive problema de desempenho com ele, que seja perceptível.

**Comment by RuanAragao** [comment]
Utilizo Zsh com [Oh My Zsh](http://ohmyz.sh/) + [Tmux](https://tmux.github.io/) no Terminator. Com eles meu workflow fica bem mais ágil, muitos comando que ajudam bastante e visual bem melhor aumentando a percepção, atalhos, integração visual com git, plugins...

Sobre desempenho, não tenho uma perda perceptível com a configuração atual, e minha maquina é bem modesta.

[Tmux Cheat Sheet](https://tmuxcheatsheet.com/)
[Oh My Zsh Cheatsheet](https://github.com/robbyrussell/oh-my-zsh/wiki/Cheatsheet)

**Comment by nathpaiva** [comment]
Eu tinha um bash super configurado cheio de coisas, mas coloquei o zsh.

Pra mim é bem tranquilo quanto ao desempenho como não faço questão de ficar configurando as coisas e não perco desempenho, eu gosto bastante dele!

**Comment by JoseCage** [comment]
ZSH  ✔️ ❤️ 

Gosto dele principalmente pelos excelentes plugins e temas disponíveis.

Minha setup básica é:

Tema: **agnosterzak** 😍
Plugins: **git, laravel, composer, gitflow, gulp, rails, themes, wp-cli, heroku, ng, vagrant, docker-compose, yarn** entre outros que não lembro.. Adoro autocomplete no teminal hehehe..

**Comment by mtmr0x** [comment]
Já chegaram aqui falando que usa `zsh` com as pedras na mão de "não sinto a diferença de performance" hahahahaha até parece que o @LFeh me esperava nessa issue haha.

O dia de largar o `zsh` chega pra muita gente, tudo no seu tempo, ele realmente tem uma latência maior ali bem visível até, mas ele realmente faz umas magias que pode acabar ajudando.

To no `bash` faz uma cota, principalmente como acesso e uso diversos servidores e computadores, nem sempre o `zsh` tá lá pra rezar a missa, o `bash` acaba servindo melhor.

Melhor é relativo.

Fialho a gente te ama, mesmo usando `zsh` :heart: (brincadeira, gente, leva a mal nan)

1 bjo

**Comment by felipefialho** [comment]
HAHAHAHAHA, JAMAIS estaria te esperando com pedras na mão, entre a gente é só ❤️ 

Mas você sente essa diferença de performance também para atividades mais "leves" Front-End, ou para aplicações mais pesadas?

Realmente não percebo.

**Comment by mtmr0x** [comment]
@LFeh eu uso o terminal para tudo, não tem uma leve criação de arquivo ou visualização de calendário que não estou no terminal, logo eu percebo bem a diferença de latencia. Ele todo é mais lentinho, pq ele é super completo e roda os processos dele quando tu vai digitando.

Fora que o que mais me irrita é que ele quer ajudar demais, esses autocomplete e esconder log depois de apertar `q` dele me deixa maluco das ideias.

**Comment by adrianoavelino** [comment]
Será que só eu aqui que sou fã do [fish](https://fishshell.com/) aqui? Acho bem simples configurar temas com [Oh my fish](https://github.com/oh-my-fish/oh-my-fish), além das configurações com o comando `fish_config`. Já experimentei o ZSH, mas me adaptei muito mais rápido com o fish.

**Comment by ghost** [comment]
Bom dia 
Sinceramente, eu procurei tentar entender a diferença entre os 2 e não notei diferença, possa ser que realmente haja alguma diferença mesmo, mas ainda não estou bom o suficiente pra perceber isso. No bash eu configurei as cores e a fonte com o tema solarized dark e o tilix como terminal.

**Comment by mstuttgart** [comment]
Eu já utilizei ambos (uso o terminal diariamente). Em questão de desempenho, o zsh puro eu não cheguei a usar. Mas utilizá-lo com 'oh my zsh' apesar dos ótimos recursos que ele trás, ficava visivelmente mais lento. Pra contornar isso eu troquei para o 'zprezto' que é um conjunto de plugins semelhante ao 'oh-my-zsh', com menos recursos porém muito mais rápido. 

Atualmente eu utilizo o bash com algumas configurações adicionais (mostrar nome da branch, ativar virtualenv quando ele está presente no diretorio - [dotfiles aqui](https://github.com/mstuttgart/dotfiles)). Decidi usa-lo por ele vir já como default em muitas distribuições linux.
